stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/TextFieldUI.kt

```diff
@@ -152,55 +157,57 @@ fun TextField(
 
     LaunchedEffect(fieldState) {
         // When field is in focus and full, move to next field so the user can keep typing
-        if (fieldState == TextFieldStateConstants.Valid.Full && hasFocus) {
+        if (fieldState == TextFieldStateConstants.Valid.Full && hasFocus.value) {
             focusManager.moveFocus(nextFocusDirection)
         }
     }
 
     val autofillReporter = LocalAutofillEventReporter.current
 
+    var selection by remember {
+        mutableStateOf<TextRange?>(null)
+    }
+
     TextFieldUi(
-        value = value,
+        value = TextFieldValue(
+            text = value,
+            selection = selection ?: TextRange(value.length),
+        ),
         loading = loading,
         onValueChange = { newValue ->
-            val acceptInput = fieldState.canAcceptInput(value, newValue)
+            val newTextValue = newValue.text
+
+            if (newTextValue == value) {
```

Comment:
I can find these nested if statements a bit hard to reason about -- can you restructure this a bit? e.g. maybe something like:
```
if (newTextValue == value || acceptInput) {
  selection = newValue.selection
}
if (acceptInput) {
  val newTextState = ...
  if (newTextState != null) {
    ... 
  }
}
```